bin/foreman-node

```diff
@@ -61,15 +65,23 @@ def plain_grains(minion)
   # We have to get the grains from the cache, because the client
   # is probably running 'state.highstate' right now.
 
-  result = IO.popen(['salt-run', '-l', 'quiet', '--output=json', 'cache.grains', minion]) do |io|
-    io.read
-  end
+  if SETTINGS[:filecache]
```

Comment:
I would prefer to have 2 small methods:
- one for getting the data via IO popen
- one using the new approach by using the data.p

Reply:
Done, latest commit uses seperate methods but keeps plain_grains to avoid code duplication

Reply:
Nice, thanks!